Samples/BlazorSampleApp/ILGPUWebHost/ComputeHostExtension.cs

```diff
@@ -0,0 +1,46 @@
+﻿// ---------------------------------------------------------------------------------------
+//                                    ILGPU Samples
+//                           Copyright (c) 2022 ILGPU Project
+//                                    www.ilgpu.net
+//
+// File: ComputeHostExtension.cs
+//
+// This file is part of ILGPU and is distributed under the University of Illinois Open
+// Source License. See LICENSE.txt for details.
+// ---------------------------------------------------------------------------------------
+
+
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using ILGPU;
+using ILGPU.Runtime;
+
+namespace BlazorSampleApp.ILGPUWebHost
+{
+
+    public static class ComputeHostExtension
+    {
+        /// <summary>
+        /// We rank accelerators by their maximum throughput.
+        /// 
+        /// Rank is approximated by clock rate * threads per processor * multiprocessors, for an integrated GPU 
+        /// we assume the clock rate is 2GHz to avoid operating specific calls
+        /// </summary>
+        /// <param name="accelerator"></param>
+        /// <returns></returns>
+        public static long AcceleratorRank(this Accelerator accelerator)
+        {
+            long clockRate = Convert.ToInt64(int.MaxValue); // assume 2 GHz for a CPU.
+
+            clockRate = (accelerator as ILGPU.Runtime.OpenCL.CLAccelerator)?.ClockRate ?? clockRate;
```

Comment:
question: are we sure that these values actually live in the same interval?

Reply:
Good question... perhaps a feature to add? This is a best guess.

Reply:
Let's leave it as it is for now 👍